tests/unit/backend/layers/api/test_portal_api.py

```diff
@@ -1813,6 +1813,134 @@ def test__dataset_meta__404(self):
         response = self.app.get(test_url_404, headers)
         self.assertEqual(response.status_code, 404)
 
+    def test__explorer_portal_integration(self):
+        """
+        Tests the explorer <-> portal integration. It works like that:
+        1. Generate the explorer_url
+        2. Call the `get_dataset_identifiers` endpoint, retrieve `collection_id` and `dataset_id` from there
+        3. Call the GET /collections/:collection_id endpoint, locate the dataset
+        """
+        headers = {"host": "localhost", "Content-Type": "application/json"}
+
+        with self.subTest("Dataset belonging to an unpublished collection"):
+
+            test_uri = "some_uri_0"
+
+            dataset = self.generate_dataset(
+                artifacts=[DatasetArtifactUpdate(DatasetArtifactType.CXG, test_uri)],
+            )
+            # In this case, explorer_url points to the canonical link
+            explorer_url = f"http://base.url/{dataset.dataset_id}.cxg/"
```

Comment:
there is a lot of duplicate code blocks that could be factored out to make reading easier. You could create a helper function that takes the `dataset` as the parameter.